python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+"""SignalExchangeAPI implementation for Tech Against Terrorism Hash List API"""
+
+
+from functools import lru_cache
+import json
+import tempfile
+import requests
+import time
+import typing as t
+from dataclasses import dataclass, field
+
+from threatexchange.exchanges.clients.techagainstterrorism import api
+
+from threatexchange.exchanges import fetch_state as state
+from threatexchange.exchanges import signal_exchange_api
+from threatexchange.exchanges import auth
+from threatexchange.exchanges.collab_config import (
+    CollaborationConfigBase,
+    CollaborationConfigWithDefaults,
+)
+from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.signal_type.pdq.signal import PdqSignal
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.signal_type.url import URLSignal
+from threatexchange.signal_type.raw_text import RawTextSignal
+
+
+@dataclass
+class TATCheckpoint(state.FetchCheckpointBase):
+    """
+    Tech Against Terrorism Hash List revolves around fetching
+    a JSON file which contains a list of hashes. The pre-signed URL
+    for the JSON file is returned from the API response.
+
+    The Hash List is updated nightly and new hashes are appended to the list.
+
+    For more information on our collection process please visit: https://terrorismanalytics.org/about/how-it-works
+    For our Hash List documentation: https://terrorismanalytics.org/docs/hash-list-v1
+    """
+
+    last_fetch_time: int
+
+    def is_stale(self) -> bool:
```

Comment:
Does this ever truly stale out though? 

Some API's, like threatexchange have time-limited deltas, which if you don't fetch within that time, the dataset becomes invalid. It sounds like you always get the whole thing in that one file, no?

Reply:
Thinking about this slightly harder, if you don't record removals to support incremental fetch, the answer is the same here as it is for the "File" signal type API - the fetch is always stale!

Reply:
Our hash list doesn't support an incremental fetch. So yes I think we can assume the fetch is always stale. In that case I will return `True`